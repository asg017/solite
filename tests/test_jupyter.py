from conftest import escape_ansi_codes


def test_hello(solite_kernel, snapshot):
    reply, output_msgs = solite_kernel.execute("select 1 + 1")
    assert (
        escape_ansi_codes(output_msgs[0]["content"]["data"]["text/plain"]) == snapshot()
    )
    assert output_msgs[0]["content"]["data"]["text/html"] == snapshot()

    reply, output_msgs = solite_kernel.execute("select 1 + ")
    assert output_msgs[0]["content"] == snapshot(name="error contents: incomplete")

    reply, output_msgs = solite_kernel.execute("select substr();")
    assert output_msgs[0]["content"] == snapshot(name="error contents: wrong num args")



# TODO
# - test JSON, numbers, null, etc.
# - error messages
# - test is_complete() messages
# - test banner?
