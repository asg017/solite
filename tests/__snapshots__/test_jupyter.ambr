# serializer version: 1
# name: test_hello
  '''
  ┌───────┐
  │ 1 + 1 │
  ├───────┤
  │     2 │
  └───────┘
  
  '''
# ---
# name: test_hello.1
  '''
  <div>
  <style>td {text-align: right;}</style><table>
  <thead>
  <tr style="text-align: center;">
  <th>
  1 + 1
  </th>
  </tr>
  </thead>
  <tbody>
  <tr>
  <td >
  2
  </td>
  </tr>
  </tbody>
  </table>
  <div>
  1 column × 1 row
  </div>
  </div>
  
  '''
# ---
# name: test_hello[error contents: incomplete]
  dict({
    'ename': 'SQL logic error',
    'evalue': '''
      error[1]: incomplete input
      
  
    ''',
    'traceback': list([
    ]),
  })
# ---
# name: test_hello[error contents: wrong num args]
  dict({
    'ename': 'SQL logic error',
    'evalue': '''
      error[1]: SQL logic error
        ┌─ TODO:1:8
        │
      1 │ select substr();
        │        ^^^^^^ wrong number of arguments to function substr()
      
  
    ''',
    'traceback': list([
    ]),
  })
# ---
# name: test_multiple
  '''
  ┌────┬────┬────┐
  │ a  │ b  │ c  │
  ├────┼────┼────┤
  │ 99 │ 99 │ 99 │
  │ 88 │ 88 │ 88 │
  │ 77 │ 77 │ 77 │
  │ 66 │ 66 │ 66 │
  │ 55 │ 55 │ 55 │
  └────┴────┴────┘
  
  '''
# ---
# name: test_multiple.1
  '''
  ┌────┬────┬────┐
  │ a  │ b  │ c  │
  ├────┼────┼────┤
  │ 99 │ 99 │ 99 │
  │ 88 │ 88 │ 88 │
  │ 77 │ 77 │ 77 │
  │ 66 │ 66 │ 66 │
  │ 55 │ 55 │ 55 │
  └────┴────┴────┘
  
  '''
# ---
